src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1922,6 +1946,17 @@ RouterSelectQuery(Query *originalQuery, RelationRestrictionContext *restrictionC
 			workerList = lappend(workerList, dummyPlacement);
 		}
 	}
+	else
+	{
+		/*
+		 * For INSERT ... SELECT,this query is still router plannable.
```

Comment:
I wouldn't put it that way, sounds too much like the whole query is router plannable which we don't know here.